### ISSUE TITLE
Q: Repository Layout

### ISSUE THREAD

**Issue by tomjn** [onelogin/wordpress-saml#46]
I was considering submitting a pull request for a `composer.json`, which would make `version.json` obsolete and bring proper attribution to `composer` based builds including this plugin.

But I see the plugin is in a subfolder rather than the root, is this due to a build process or continuous integration system at onelogin? Or a custom dependency manager?

**Comment by pitbulk** [comment]
I agree that some customers that use composer will prefer a "composer based plugin".
We created that plugin years ago, when the use of composer was minor, and that why the current plugin has that structure.

**Comment by nicoladj77** [comment]
@pitbulk could we consider changing the structure?I would need to use this as a Git submodule on a VIP Go install and the current Layout doesn't allow this.

**Comment by pitbulk** [comment]
We can create a new branch, and do all the required changed that you request, I will try to maintain both branches (master and composer branch)  updated with the new changes/features.

**Comment by pitbulk** [comment]
@tomjn are you able to create that new composer branch?

**Comment by tomjn** [comment]
I've been without internet for the last week, but completing these commands should be enough:

```
git checkout -b composer
git mv onelogin-saml-sso/* .
git commit -m "move plugin files one folder up"
composer init --name="onelogin/wordpress-saml" --type="wordpress-plugin" --license="MIT"
git add composer.json
git commit -m "added composer.json"
git push origin composer
```

**Comment by pitbulk** [comment]
I added the [composer branch](https://github.com/onelogin/wordpress-saml/tree/composer), but it will depend on [php-saml 3.0.0](https://github.com/onelogin/php-saml/tree/3.0.0) that is still under development (need to make it PSR-4 and PSR-7 before an official release)

Meanwhile, if you need to install wordpress-saml using composer, read that [article](https://roots.io/using-composer-with-wordpress/), you will see that there is a "composer version" of wordpress-saml at
wpackagist-plugin/onelogin-saml-sso

**Comment by sandykadam** [comment]
@pitbulk This is really needed now since there was outage of this plugin from Wordpress/packagist site, we are now planning to have direct github zip URL to download via composer, but because of the folder structure its not possible to implement. 
So update of folder structure or different branch updated with latest changes will really appreciated.

Thanks

**Comment by pitbulk** [comment]
The plugin is back on the directory

**Comment by sandykadam** [comment]
Yes I know its back that's why I closed my other issue, but I'm looking for alternate solution and not to depend on Wordpress download or wpackagist to download plugin.

I'm planning to add direct github URL in composer but due to folder structure its not possible.

**Comment by pitbulk** [comment]
Right now I don't have much time to spend, but will give to this priority when available
